### ISSUE TITLE
yan.lock(v1) is invalid yaml; Support Serialization to: Yaml & JSON

### ISSUE THREAD

**Issue by hsjobeki** [oven-sh/bun#5787]
### What is the problem this feature would solve?

I'd like to parse the yarn lockfile.
Currently this is very complicated because yarn.lockfile (V1) is invalid yaml.

### What is the feature you are proposing to solve the problem?

I'd like to just parse the lockfile in a simple and reliable way.  For me this could be solved by:

- Allow users to specify the serialization format "yarnv1", "yaml", "json"

### What alternatives have you considered?

Parsing yarn.lock with additional npm packages (annoying / feels flakey)
Converting yarn.lock v1 to yarn.lock v2/v3 (annoying / feels flakey)

**Comment by Electroid** [comment]
Can you elaborate a bit? Are you using `bun install --yarn`?

**Comment by hsjobeki** [comment]
Yes i'm using bun install --yarn. Problem beeing that yarn lockfile version 1 is not valid yaml. (By Yarn specification)
Therefore i cannot use a yaml parser to deserialize the yarn.lock (v1)
This makes the yarn.lock (v1) pretty hard to reuse for third party tools. 
I'd need a standardized format, where i can get an of-the-shelf parser such as yaml and/or json.
Otherwise this yarn.lock feature is only nice for git diff and nothing else.

**Comment by hsjobeki** [comment]
A little further context i evaluted bun as a node alternative. But i need to integrate it with my build processes. This requires pre calculating the dependency graph. Which is not possible because i cannot parse the lockfile(s) of bun

**Comment by hsjobeki** [comment]
@Electroid, I think the npm label you assigned doesn't match the problem.
I'd suggest a label like `lockfile` and/or `ecosystem`

**Comment by RiskyMH** [comment]
The yarn.lock output file isn't as useful now that we have `bun.lock` a JSON*C* based lockfile. Only caveat is its also not fully standard (as it has trailing comma to make diffs better).

**Comment by hsjobeki** [comment]
> The yarn.lock output file isn't as useful now that we have `bun.lock` a JSON_C_ based lockfile. Only caveat is its also not fully standard (as it has trailing comma to make diffs better).

Nice. Thanks for the fix. I think i can pipe the lockfile through prettier or another tolerant formatter to get rid of the trailing commas and get something jsonc-compliant.
